network/client.go

```diff
@@ -0,0 +1,84 @@
+package network
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+
+	"github.com/rs/zerolog"
+	"github.com/statechannels/go-nitro/network/serde"
+	"github.com/statechannels/go-nitro/network/transport"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/protocols/directdefund"
+	"github.com/statechannels/go-nitro/protocols/directfund"
+	"github.com/statechannels/go-nitro/protocols/virtualdefund"
+	"github.com/statechannels/go-nitro/protocols/virtualfund"
+)
+
+type Response struct {
```

Comment:
Nit but we could add some godoc to this file

Reply:
[b2c9f63](https://github.com/statechannels/go-nitro/pull/1102/commits/b2c9f63b3d04ac5c1dd8ed306883b2b887728d69)

Reply:
Good call. Added in [20f6122](https://github.com/statechannels/go-nitro/pull/1102/commits/20f61224c8b7d1032c40a5ba6ec910bf09be4b96).